xbmc/GUIPassword.cpp

```diff
@@ -549,30 +548,34 @@ bool CGUIPassword::IsDatabasePathUnlocked(const std::string& strPath, VECSOURCES
   return false;
 }
 
-bool CGUIPassword::IsMediaPathUnlocked(const std::string& strPath, const std::string& strType)
+void CGUIPassword::SetMediaSourcePath(const std::string& MediaSourcePath)
 {
-  if (StringUtils::StartsWithNoCase(strPath, "root") ||
-      StringUtils::StartsWithNoCase(strPath, "library://"))
+  if (!StringUtils::StartsWithNoCase(MediaSourcePath, "root") &&
+      !StringUtils::StartsWithNoCase(MediaSourcePath, "library://"))
   {
-    // no mediasource-lookup needed
-    CLog::Log(LOGDEBUG, "CGUIPassword::IsMediaPathUnlocked - entering from {}", strPath);
-    return true;
+    m_strMediaSourcePath = MediaSourcePath;
   }
+  else
+  {
+    m_strMediaSourcePath.clear();
+  }
+}
```

Comment:
No, please keep the setter as simple and inline in the header file as it was before! No additional logics in the setter, please. keep the root/library check in CGUIPassword::IsMediaPathUnlocked as it was before, please.

Reply:
> imo 072afd2 would be a valid usecase for the setter.

aaahhh, ok, thought you saw that "usecase"-commit... i'll change it back